### PR TITLE
Nano: add channels_last convert check

### PR DESCRIPTION
**Pull request by ACupofAir**
### 1. Why the change?
When we pass in a `input_sample` which is not consist of tensor, or one of the item in input doesn't have `.to()` method, it will lead to an error to convert it on `channels_last` is true.

### 2. Summary of the change 
* When user directly use `trace` or `quantize` function with a `input_sample` pass in or use `optimize` function which mean the `input_sample` is **not None**, we will compute a list named `channels_last_available` which instruct which item in the `input_sample` can be convert to `channels_last`.
* When user directly use `trace` or `quantize` function and don't pass in `input_sample`, make the list `channels_last_available` empty, and the `forward_step` function will convert default.
* Add unit-test of methods with channels_last

**warning**: 
* The methods with jit hasn't been handled, I'm working on it.
* This pr is to replace #6918 for the previous pr conflicts with main branch

### CODE REVIEW

**Review by ACupofAir — COMMENTED** [review]
LGTM

File: python/nano/src/bigdl/nano/deps/ipex/ipex_inference_model.py [hunk 1]
```diff
@@ -74,6 +76,12 @@ def __init__(self, model: torch.nn.Module, input_sample=None, use_ipex=False, dt
         self.original_model = model
         if self.channels_last:
             self.model = self.model.to(memory_format=torch.channels_last)
+            if channels_last_available: # init from _load, the channels_last_available is not none
+                self.channels_last_available = channels_last_available
+            else:
+                self.channels_last_available = generate_channels_last_available(input_sample)
+        else:
+            self.channels_last_available = []
```
Comment (TheaperDeng) [inline python/nano/src/bigdl/nano/deps/ipex/ipex_inference_model.py:84]: add same logic to if load code block for jit related model

File: python/nano/src/bigdl/nano/utils/inference/pytorch/model_utils.py [hunk 2]
```diff
@@ -198,6 +198,26 @@ def get_input_example(model, input_sample, forward_args):
...
+    '''
+    # try channels_last available
+    if inputs:  # to avoid the situation of inputs == None
+        channels_last_available = [True] * len(inputs)
+        for idx, input in enumerate(inputs):
+            try:
+                input.to(memory_format=torch.channels_last)
+            except Exception as _e:
+                channels_last_available[idx] = False
+            else:
```
Comment (rnwang04) [inline python/nano/src/bigdl/nano/utils/inference/pytorch/model_utils.py:214]: no need for this else

File: python/nano/test/pytorch/tests/inference/test_ipex_jit_inference.py [hunk 3]
```diff
@@ -103,6 +113,49 @@ def test_ipex_jit_inference(self):
         with InferenceOptimizer.get_context(new_model):
             new_model(self.data_sample)
     
+    def test_ipex_channels_last_inference(self):
```
Comment (rnwang04) [inline python/nano/test/pytorch/tests/inference/test_ipex_jit_inference.py:116]: One suggestion here, may not change your code.
For this case, maybe just define one ut like `test_jit_ipex_channels_last_inference` here and list three cases inside the the function, so you don't need to define model, x1, x2, x3 three times.

**Review by rnwang04 — APPROVED** [review]
others LGTM

**Pull request merged**